library/std/src/sys/unix/process/process_common.rs

```diff
@@ -224,6 +229,9 @@ impl Command {
     pub fn get_gid(&self) -> Option<gid_t> {
         self.gid
     }
+    pub fn get_groups(&self) -> Option<&[gid_t]> {
+        if self.groups.is_empty() { None } else { Some(&self.groups) }
```

Comment:
This should not treat "empty" and "not provided" identically.

Reply:
Hmm I guess I'm not sure how to proceed since @shepmaster said it couldn't be Option<Box<[gid_t]>>. I'm new to Rust, so I wasn't sure what he meant there.

Reply:
@slo1 That was referring to the return value of this method, not the `self.groups` type. The `groups` field can use `Option<Box<[gid_t]>>`, I think, and then this method can turn that into `Option<&[gid_t]>` and return it. You just shouldn't be checking `is_empty`, because "set an empty group list" is valid and distinct from "don't set a group list".